src/irmin-pack/unix/ext.ml

```diff
@@ -128,100 +124,92 @@ module Maker (Config : Conf.S) = struct
           node : read Node.CA.t;
           commit : read Commit.CA.t;
           branch : Branch.t;
-          index : Index.t;
-          dict : Pack_dict.t;
-          io : Io_legacy.t;
+          fm : File_manager.t;
+          dict : Dict.t;
         }
 
-        let contents_t t : 'a Contents.t = t.contents
-        let node_t t : 'a Node.t = (contents_t t, t.node)
-        let commit_t t : 'a Commit.t = (node_t t, t.commit)
-        let branch_t t = t.branch
-
         let batch t f =
-          Commit.CA.batch t.commit (fun commit ->
-              Node.CA.batch t.node (fun node ->
-                  Contents.CA.batch t.contents (fun contents ->
-                      let contents : 'a Contents.t = contents in
-                      let node : 'a Node.t = (contents, node) in
-                      let commit : 'a Commit.t = (node, commit) in
-                      f contents node commit)))
-
-        let unsafe_v config =
-          let root = Conf.root config
-          and fresh = Conf.fresh config
-          and lru_size = Conf.lru_size config
-          and readonly = Conf.readonly config
-          and log_size = Conf.index_log_size config
-          and throttle = Conf.merge_throttle config
-          and indexing_strategy = Conf.indexing_strategy config in
-          let f = ref (fun () -> ()) in
-          let index =
-            Index.v
-              ~flush_callback:(fun () -> !f ())
-                (* backpatching to add pack flush before an index flush *)
-              ~fresh ~readonly ~throttle ~log_size root
-          in
-          let dict =
-            let path = Irmin_pack.Layout.dict ~root in
-            Dict.v ~fresh ~readonly path
+          let contents = Contents.CA.cast t.contents in
+          let node = Node.CA.Pack.cast t.node in
+          let commit = Commit.CA.cast t.commit in
+          let contents : 'a Contents.t = contents in
+          let node : 'a Node.t = (contents, node) in
+          let commit : 'a Commit.t = (node, commit) in
+          let on_success res =
+            File_manager.flush_exn t.fm;
+            Lwt.return res
           in
-          let io =
-            let path = Irmin_pack.Layout.pack ~root in
-            (* If the file already exists in V1, we will bump the generation header
-                   lazily when appending a V2 entry. *)
-            let version = Some Irmin_pack.Version.latest in
-            Io_legacy.v ~version ~fresh ~readonly path
+          let on_fail exn =
+            [%log.info "[pack] flush during batch fail"];
```

Comment:
I'm not sure I understand this - `on_fail` is called when `f` fails, but `f` might not fail because of a flush. The message seems to say that its necessarily a flush that failed.

Reply:
I've changed the messages to the following:
```
[%log.info "[pack] batch failed. calling flush"];
[%log.err "[pack] batch failed and flush failed. Silencing flush fail"];
```